nova_vm/src/ecmascript/types/language/object/property_key.rs

```diff
@@ -122,9 +122,26 @@ impl TryFrom<Value> for PropertyKey {
     fn try_from(value: Value) -> Result<Self, Self::Error> {
         match value {
             Value::Integer(x) => Ok(PropertyKey::Integer(x)),
+            Value::Float(x)
```

Comment:
**issue:** This should be unnecessary: The Venn diagram of the three different Number variants should have no overlap, meaning that for any `Value::Float(x)` there cannot be a **lossless** function `f` for which `Value::Integer(f(x))` is a valid `Value::Integer` value, and vice versa. Same of course goes for heap numbers.

So given any numeric value N there is only ever one proper Number variant for it.

Thus, for every `Value::Float(x)` it is always trivially true that `x` cannot be represented as a non-negative integer.

Reply:
I think -0 is not actually a proper indexed integer. The spec is a bit vague on this, -0 is a canonical numeric string but it's not a valid index.

So I think checking for it is not necessary. Need to probably check the spec with a bit finer-toothed a comb to be really sure though.

Reply:
Yeah, this definitely seems to be how it works. I wonder why on earth the spec has the special casing for `-0`... Presumably it pops up _somewhere_ as some kind of a contradictory special case...